diff --git a/node_modules/playwright/lib/common/index.js b/node_modules/playwright/lib/common/index.js
index ace2c1a..3282ec6 100644
--- a/node_modules/playwright/lib/common/index.js
+++ b/node_modules/playwright/lib/common/index.js
@@ -869,7 +869,8 @@ function resolve(specifier, context, nextResolve) {
     const filename = import_url.default.fileURLToPath(context.parentURL);
     const resolved = resolveHook(filename, specifier);
     if (resolved !== void 0) {
-      specifier = context.conditions?.includes("import") ? import_url.default.pathToFileURL(resolved).toString() : resolved;
+      const isImport = context.conditions?.has?.("import") ?? context.conditions?.includes?.("import");
+      specifier = isImport ? import_url.default.pathToFileURL(resolved).toString() : resolved;
     }
   }
   const result2 = nextResolve(specifier, context);
diff --git a/node_modules/playwright/lib/transform/esmLoader.js b/node_modules/playwright/lib/transform/esmLoader.js
index 2ae509c..db1dcfe 100644
--- a/node_modules/playwright/lib/transform/esmLoader.js
+++ b/node_modules/playwright/lib/transform/esmLoader.js
@@ -5629,7 +5629,8 @@ function resolve(specifier, context, nextResolve) {
     const filename = import_url.default.fileURLToPath(context.parentURL);
     const resolved = resolveHook(filename, specifier);
     if (resolved !== void 0) {
-      specifier = context.conditions?.includes("import") ? import_url.default.pathToFileURL(resolved).toString() : resolved;
+      const isImport = context.conditions?.has?.("import") ?? context.conditions?.includes?.("import");
+      specifier = isImport ? import_url.default.pathToFileURL(resolved).toString() : resolved;
     }
   }
   const result2 = nextResolve(specifier, context);
